### DISCUSSION TITLE
Processing won't continue after finishing file download

### DISCUSSION THREAD

**Discussion by mpemburn** [maennchen/ZipStream-PHP#184] [Q&A] [answered]
I just added ZipStream to my Laravel 8.x project today and it works pretty darned well.  However, when it gets finished with downloading the stream, it just hangs there.  Only if I manually refresh the page will it continue.  Here's my code:

```php
        $options = new Archive();
        $options->setSendHttpHeaders(true);
        $options->setEnableZip64(false);

        $filepath = 'download.' . date('Y.m.d.h.i.s') . '.zip';

        $zip = new ZipStream($filepath, $options);

        $count = 0;
        foreach ($files as $file) {
            $fileName = $file->name;
            $content = file_get_contents($file->signedUrl());
            $zip->addFile($fileName, $content);

            $count++;
        }

        try {
            $zip->finish();
        } catch (OverflowException $e) {
            flash()->error('Error: A problem occurred with zip streaming.');

            return back();
        }

        flash($count . ' file(s) zipped and downloaded.');

        return back();


```

Thanks in advance for any help you can offer!

Mark Pemburn
Trak Software, Inc.

**Comment by NicolasCARPi** [comment, marked as answer]
Hello,

What's `back()` ?

**Reply by mpemburn** [reply]
It's a Laravel helper function that redirects to the calling page.

**Reply by maennchen** [reply]
@mpemburn I don't think that you can redirect a user on the same request that serves a zip file.

You would probably try to open the zip file link in a new tab to make sure the browser doesn't navigate there.

Something like 

```html
<a href="link-to-zip" download="name.zip" target="_blank">Download Zip</a>
```

**Reply by mpemburn** [reply]
Thanks, I'll give that a try!
